airflow/ui/src/main.tsx

```diff
@@ -58,6 +59,9 @@ axios.interceptors.request.use((config) => {
   return config;
 });
 
+// Dynamically set the base URL for XHR requests based on the meta tag.
+OpenAPI.BASE = document.querySelector("meta[name='backend-server-base-url']")?.getAttribute("content") ?? "";
```

Comment:
How is this meta tag put in to the HTML? Does it already exist?

Reply:
It does not. You need to manually add it inside the `index.html` which is a 20 line file atm.

There are more details in the `How to test` section of the PR.

Reply:
Could we use Jinja templating here for index.html, at least for the "prod" build? Something like this would allow fastapi to serve up the right html dynamically based on a config variable:
```
<!doctype html>
<html lang="en" style="height: 100%">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/png" href="/webapp/pin_32.png" />
    <meta name="backend-server-base-url" content="{{ base_url }}">
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Airflow 3.0</title>
    <script type="module" crossorigin src="/static/assets/index-DtWM-pqV.js"></script>
  </head>
  <body style="height: 100%">
    <div id="root" style="height: 100%"></div>
  </body>
</html>
```

Reply:
Oh yeah - or do I mis-understand otherwise... if I operate my prod site in a subfolder I'd need to patch a html file inside site-packages to adjust the path? Temporarily that might be OK but for 3.0 this needs to be configurable from outside w/o patching files.

Reply:
Yeah, we need an option that doesn't require changes to Airflow in the image for this to be complete -- it was possible (and widely) used on Airflow 2, we can't have every person who needs this to touch a file in the airflow install to get this ability back.

Reply:
I’ll have to check.

At the moment the Index file comes from the vite build process. I don’t think we can go around that.

Maybe a python script or helper to manipulate this index file ?

Reply:
Updated, should look better.

Basically the index file is templated and the meta tag will always be present taking the config value for `fastapi.base_url`. (default to `localhost:29091` but can be replaced with anything else, `locahost:9091` in my example, we can't add a prefix yet such as `locahost:29091/d12345` because static file resolution will fail until we do https://github.com/apache/airflow/issues/46548)

Reply:
The update is much better. I'm glad we can make the airflow config work instead.